example/callback.cc

```diff
@@ -3,135 +3,90 @@
 #include <cstdlib>
 #include <string>
 #include <cinttypes>
+#include <variant>
 
-#include "wasm.hh"
+#include "imaginary-wasm.hh"
 
+namespace {
 // Print a Wasm value
-auto operator<<(std::ostream& out, const wasm::Val& val) -> std::ostream& {
-  switch (val.kind()) {
-    case wasm::ValKind::I32: {
-      out << val.i32();
-    } break;
-    case wasm::ValKind::I64: {
-      out << val.i64();
-    } break;
-    case wasm::ValKind::F32: {
-      out << val.f32();
-    } break;
-    case wasm::ValKind::F64: {
-      out << val.f64();
-    } break;
-    case wasm::ValKind::ANYREF:
-    case wasm::ValKind::FUNCREF: {
-      if (val.ref() == nullptr) {
-        out << "null";
-      } else {
-        out << "ref(" << val.ref() << ")";
-      }
-    } break;
-  }
+auto operator<<(std::ostream &out, const wasm::Val &val) -> std::ostream& {
+  std::visit([](auto &&arg) { std::cout << arg; });
   return out;
 }
 
 // A function to be called from Wasm code.
-auto print_callback(
-  const wasm::vec<wasm::Val>& args, wasm::vec<wasm::Val>& results
-) -> wasm::own<wasm::Trap> {
-  std::cout << "Calling back..." << std::endl << "> " << args[0] << std::endl;
-  results[0] = args[0].copy();
-  return nullptr;
+int32_t print_callback(int32_t v) {
+  std::cout << "Calling back..." << std::endl << "> " << v << std::endl;
+  return v;
```

Comment:
This will prevent returning multi-values I think.

Reply:
I was thinking multi-value returns would look like this:
```
std::tuple<int32_t, float> my_callback() {
  return {1, 2.0};
}
```
which when called from normal C++ would look like:
```
void foo() {
    auto [x, y] = my_callback();
}
```
Ideally we want that same syntax when C++ code calls a multi-value return wasm function.